android-module.gradle

```diff
@@ -10,24 +11,28 @@ android {
     }
     buildTypes {
         debug {
-            sourceSets {
-                main.java.srcDirs += 'build/generated/ksp/debug/kotlin/'
-            }
         }
         release {
-            sourceSets {
-                main.java.srcDirs += 'build/generated/ksp/release/kotlin/'
-            }
+        }
+    }
+
+    // KSP 2.0 source set configuration
```

Comment:
Added proper KSP 2.0 source set configuration outside of buildTypes. This ensures correct source set isolation between debug and release variants, preventing the "Overload resolution ambiguity" errors that occur when both variants see each other's generated code.